lib/utils.js

```diff
@@ -12,21 +17,72 @@ var basicAuth = require('basic-auth');
  * @param   {string}   password Expected password
  * @returns {function} Express 4 middleware requiring the given credentials
  */
-exports.basicAuth = function(username, password) {
-	return function(req, res, next) {
-
-		if (!username || !password) {
-			console.log('Username or password is not set.');
-			return res.send('<h1>Error:</h1><p>Username or password not set. <a href="https://github.com/alphagov/govuk_prototype_kit/blob/master/docs/deploying.md#3-set-a-username-and-password">See guidance for setting these</a>.</p>');
-		}
-		
-		var user = basicAuth(req);
-
-		if (!user || user.name !== username || user.pass !== password) {
-			res.set('WWW-Authenticate', 'Basic realm=Authorization Required');
-			return res.sendStatus(401);
-		}
-
-		next();
-	};
+function basicAuthUsernamePassword (username, password) {
+  return function(req, res, next) {
+
+    if (!username || !password) {
+      console.log('Username or password is not set.');
+      return res.send('<h1>Error:</h1><p>Username or password not set. <a href="https://github.com/alphagov/govuk_prototype_kit/blob/master/docs/deploying.md#3-set-a-username-and-password">See guidance for setting these</a>.</p>');
+    }
+
+    var user = basicAuth(req);
+
+    if (!user || user.name !== username || user.pass !== password) {
+      res.set('WWW-Authenticate', 'Basic realm=Authorization Required');
+      return res.sendStatus(401);
+    }
+
+    next();
+  };
+}
+
+// Checks all modules in package.json are available, quit if not available
+function checkNodeModules(){
+  var env = process.env.NODE_ENV || 'development',
+      nodeModules = require(__dirname + '/../package.json');
```

Comment:
If you're using `/../` in a path string, it's best to use `path.join` again as it will clean up the string properly.